version: '3.7'

services:
  python:
    build:
      context: .
      dockerfile: deploy/docker/python/Dockerfile
    environment:
      AWS_REGION: us-west-2
      S3_PATH: s3://demon-backup/test-backup
      RESTORE_FORCE: "true"
      RESTORE_DATE: latest # "2019-09-21T17:53:29Z"
      RESTORE_DELETE: "true"
      DATA_DIRECTORY: /data
    volumes:
      - ./s3-backup-restore/entrypoint.sh:/home/appuser/entrypoint.sh
      - ./s3-backup-restore/run.sh:/home/appuser/run.sh
      - ./test-data:/data
    secrets:
      - AWS_ACCESS_KEY
      - AWS_SECRET_KEY

secrets:
  AWS_ACCESS_KEY:
    file: ./deploy/docker/dev-secrets/AWS_ACCESS_KEY
  AWS_SECRET_KEY:
    file: ./deploy/docker/dev-secrets/AWS_SECRET_KEY
